Flixel Features/SetTileProperties/source/PlayState.hx

```diff
@@ -0,0 +1,134 @@
+package;
+
+import flixel.addons.editors.ogmo.FlxOgmoLoader;
+import flixel.effects.particles.FlxEmitter;
+import flixel.effects.particles.FlxParticle;
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.FlxSprite;
+import flixel.FlxState;
+import flixel.text.FlxText;
+import flixel.tile.FlxTile;
+import flixel.tile.FlxTilemap;
+import flixel.ui.FlxButton;
+import flixel.math.FlxMath;
+import flixel.util.FlxColor;
+
+class PlayState extends FlxState
+{
+	
+	private var _map:FlxTilemap;
+	private var _leftEmitter:FlxTypedEmitter<FlxParticle>;
+	private var _rightEmitter:FlxTypedEmitter<FlxParticle>;
+	
+	private var _scores:Array<Int> = [0, 0];
+	
+	private static inline var RED:Int = 0;
+	private static inline var BLUE:Int = 1;
+	
+	private var _txtScoreRed:FlxText;
+	private var _txtScoreBlue:FlxText;
+	
+	
+	override public function create():Void
+	{
+		bgColor = 0xff331133;
+		
+		super.create();
+		
+		var mapRow:String = "0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 2, 2, 2, 2, 2, 0, 0\n\r";
+		var mapData:String='';
+		for (i in 0...Std.int(FlxG.height / 16))
+		{
+			mapData += mapRow;
+		}
+		
+		_map = new FlxTilemap();
+		_map.loadMapFromCSV(mapData, AssetPaths.tiles__png, 16, 16);
+		
+		_map.setTileProperties(1, FlxObject.RIGHT, leftHit); // Tile #1 will only collide on it's right-side, and will call 'leftHit' when it does.
+		_map.setTileProperties(2, FlxObject.LEFT, rightHit); // Tile #2 will only collide on it's left-side, and will call 'rightHit' when it does.
+		
+		add(_map);
+		
+		_leftEmitter = new FlxTypedEmitter<FlxParticle>(0, Std.int(FlxG.height * .6), 50);
```

Comment:
There's a lot of redundant logic in this emitter creation code, would be nicer to add a separate function for it.